gpjax/bayes_opt/acquisition_optimiser.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 The JaxGaussianProcesses Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from dataclasses import dataclass
+
+import jax.numpy as jnp
+import jax.random as jr
+from jaxopt import ScipyBoundedMinimize
+
+from gpjax.bayes_opt.acquisition_functions import AcquisitionFunction
+from gpjax.bayes_opt.search_space import (
+    AbstractSearchSpace,
+    BoxSearchSpace,
+)
+from gpjax.typing import (
+    Array,
+    Float,
+    KeyArray,
+)
+
+
+@dataclass
+class AbstractAcquisitionOptimiser(ABC):
+    """Abstract base class for acquisition function optimisers."""
+
+    @abstractmethod
+    def optimise(
```

Comment:
Design choice - do we want `optimise` or `minimise`? The latter is nice as it connect with the API in other optimisation libraries like Scipy. Further, we may wish to define `optimize` or `minimize` here to simply call the respective method. It makes the package more accessible to people using American-English.